src/slider-new/helpers/useMove.ts

```diff
@@ -0,0 +1,246 @@
+        addGlobalListener(window, "touchend", onTouchEnd, false);
+        addGlobalListener(window, "touchcancel", onTouchEnd, false);
+      };
+    } else {
+      const onPointerMove = (e: PointerEvent) => {
+        if (e.pointerId === state.current.id) {
+          // @ts-ignore
+          const pointerType: BaseMoveEvent["pointerType"] =
+            e.pointerType || "mouse";
+
+          // Problems with PointerEvent#movementX/movementY:
+          // 1. it is always 0 on macOS Safari.
+          // 2. On Chrome Android, it's scaled by devicePixelRatio, but not on Chrome macOS
+          if (isNull(state.current.lastPosition)) return;
+          move(
+            pointerType,
+            e.pageX - state.current.lastPosition.pageX,
+            e.pageY - state.current.lastPosition.pageY,
+          );
+          state.current.lastPosition = { pageX: e.pageX, pageY: e.pageY };
+        }
+      };
+
+      const onPointerUp = (e: PointerEvent) => {
+        if (e.pointerId === state.current.id) {
+          // @ts-ignore
+          const pointerType: BaseMoveEvent["pointerType"] =
+            e.pointerType || "mouse";
+          end(pointerType);
+          state.current.id = null;
+          removeGlobalListener(window, "pointermove", onPointerMove, false);
+          removeGlobalListener(window, "pointerup", onPointerUp, false);
+          removeGlobalListener(window, "pointercancel", onPointerUp, false);
+        }
+      };
+
+      moveProps.onPointerDown = (e: React.PointerEvent) => {
+        if (e.button === 0 && state.current.id == null) {
+          start();
+          e.stopPropagation();
+          e.preventDefault();
+          state.current.lastPosition = { pageX: e.pageX, pageY: e.pageY };
+          state.current.id = e.pointerId;
+          addGlobalListener(window, "pointermove", onPointerMove, false);
+          addGlobalListener(window, "pointerup", onPointerUp, false);
+          addGlobalListener(window, "pointercancel", onPointerUp, false);
+        }
+      };
+    }
+
+    const triggetKeyboardMove = (deltaX: number, deltaY: number) => {
+      start();
+      move("keyboard", deltaX, deltaY);
+      end("keyboard");
+    };
+
+    moveProps.onKeyDown = e => {
+      switch (e.key) {
+        case "Left":
+        case "ArrowLeft":
+          e.preventDefault();
+          e.stopPropagation();
+          triggetKeyboardMove(-1, 0);
+          break;
+        case "Right":
+        case "ArrowRight":
+          e.preventDefault();
+          e.stopPropagation();
+          triggetKeyboardMove(1, 0);
+          break;
+        case "Up":
+        case "ArrowUp":
+          e.preventDefault();
+          e.stopPropagation();
+          triggetKeyboardMove(0, -1);
+          break;
+        case "Down":
+        case "ArrowDown":
+          e.preventDefault();
+          e.stopPropagation();
+          triggetKeyboardMove(0, 1);
+          break;
+      }
```

Comment:
Can we use a fallthrough switch case?

```js
      switch (e.key) {
        case "Up":
        case "Left":
        case "Down":
        case "Right":
        case "ArrowUp":
        case "ArrowLeft":
        case "ArrowDown":
        case "ArrowRight":
          e.preventDefault();
          e.stopPropagation();
        case "Left":
        case "ArrowLeft":
          triggetKeyboardMove(-1, 0);
          break;
        case "Right":
        case "ArrowRight":
          triggetKeyboardMove(1, 0);
          break;
        case "Up":
        case "ArrowUp":
          triggetKeyboardMove(0, -1);
          break;
        case "Down":
        case "ArrowDown":
          triggetKeyboardMove(0, 1);
          break;
      }
```

Reply:
We can use this, looks nice !!

Reply:
I think we don't have to make these changes in our repo.
You can raise a PR directly to React Aria.